app/javascript/packages/document-capture/components/hybrid-doc-capture-warning.tsx

```diff
@@ -0,0 +1,67 @@
+import { useContext, ReactNode } from 'react';
+import { useI18n, formatHTML } from '@18f/identity-react-i18n';
+import { getConfigValue } from '@18f/identity-config';
+import ServiceProviderContext from '../context/service-provider';
+
+function formatWithStrong(text: string): ReactNode {
+  return formatHTML(text, {
+    strong: ({ children }) => <strong>{children}</strong>,
+  });
+}
+
+interface HybridDocCaptureWarningProps {
+  /**
+   * A class string to append to root element
+   */
+  className?: string;
+}
+
+function HybridDocCaptureWarning({ className = '' }: HybridDocCaptureWarningProps): JSX.Element {
+  const { t } = useI18n();
+  // Determine the Service Provider name to display,
+  // in some circumstances.
+  // If there is no SP, we default to the appName
+  const spContext = useContext(ServiceProviderContext);
+  const serviceProviderName = spContext.name;
+  const appName = getConfigValue('appName');
+
+  const listHeadingText = t('doc_auth.hybrid_flow_warning.only_add_if_text');
+  const ownAccountItemText = t('doc_auth.hybrid_flow_warning.only_add_own_account_html', {
+    app_name: appName,
+  });
+  const phoneVerifyItemText = t('doc_auth.hybrid_flow_warning.only_add_phone_verify_html', {
+    app_name: appName,
+  });
+  let spServicesItemText;
+  let warningText = t('doc_auth.hybrid_flow_warning.explanation_non_sp_html', {
+    app_name: appName,
+  });
+
+  if (serviceProviderName) {
+    warningText = t('doc_auth.hybrid_flow_warning.explanation_html', {
+      app_name: appName,
+      service_provider_name: serviceProviderName,
+    });
+    spServicesItemText = t('doc_auth.hybrid_flow_warning.only_add_sp_services_html', {
+      service_provider_name: serviceProviderName,
+    });
+  }
+
+  return (
+    <div className={`usa-alert usa-alert--warning ${className}`} role="status">
```

Comment:
Can we use the [existing `<Alert />` component](https://github.com/18F/identity-idp/blob/main/app/javascript/packages/components/alert.tsx)? Though I think you might need to add support for something similar to what was done with the [Rails version to allow customization of the "`text_tag`"](https://github.com/18F/identity-idp/blob/393294ac6a704e49ca770b31fa3076e60db4c37e/app/components/alert_component.rb#L6).

Reply:
I started with that in mind, but it's a bit more complicated, because there are `<p>` and `<ul>` elements as siblings. I suppose they could be wrapped in something like a generic `div` and then `div` could be specified as the element type somehow, but that seems counter to the USWDS guidance.
  
Perhaps we could have a prop `body=true` (or similar), which tells the component "all of the children should go directly into the `usa-alert__body` div. Otherwise, into a `<p>`. I can whip up a commit of that if you think the juice is worth the squeeze

Reply:
That's essentially what `text_tag` was meant for in the Rails implementation, since we had a similar need, and the solution was to swap the `<p>` with a `<div>`, for example.

e.g.

```tsx
<Alert textTag="div">
  <p>{/* ... */}</p>
  <u>{/* ...*/}</ul>
</Alert>
```

I also agree that, in general, the alerts should be short and not include such complex text. Another reason in favor of this is that they're implemented as ARIA live status regions, which can be announced in full to a screen reader as soon as it's added to the page.